### ISSUE TITLE
`__match_args__` must catch non-existent attributes

### ISSUE THREAD

**Issue by sobolevn** [python/mypy#13741]
Simple case:

```python
class MyClass:
    a: int
    __match_args__ = ('a', 'b')
```

I think that it should generate an error: there's no `b` attribute.

**Comment by erictraut** [comment]
This condition doesn't generate a runtime error. I could see this maybe being indicative of an unintended bug, but it's probably pretty low priority. Perhaps mark it as a feature rather than a bug?
